packages/template-retail-react-app/app/partials/product-view/index.jsx

```diff
@@ -76,175 +76,366 @@ const ButtonWithRegistration = withRegistration(Button)
  * Render a product detail view that includes name, image gallery, price,
  * variant selections, action buttons
  */
-const ProductView = ({
-    product,
-    category,
-    showFullLink = false,
-    imageSize = 'md',
-    isWishlistLoading = false,
-    addToCart,
-    updateCart,
-    addToWishlist,
-    updateWishlist,
-    isProductLoading,
-    isProductPartOfSet = false
-}) => {
-    const intl = useIntl()
-    const history = useHistory()
-    const location = useLocation()
-    const {
-        isOpen: isAddToCartModalOpen,
-        onOpen: onAddToCartModalOpen,
-        onClose: onAddToCartModalClose
-    } = useAddToCartModalContext()
-    const theme = useTheme()
-    const [showOptionsMessage, toggleShowOptionsMessage] = useState(false)
-    const {
-        showLoading,
-        showInventoryMessage,
-        inventoryMessage,
-        quantity,
-        minOrderQuantity,
-        setQuantity,
-        variant,
-        variationParams,
-        variationAttributes,
-        stockLevel,
-        stepQuantity
-    } = useProduct(product, isProductPartOfSet)
-    const canAddToWishlist = !isProductLoading
-    const canOrder =
-        !isProductLoading &&
-        variant?.orderable &&
-        parseInt(quantity) > 0 &&
-        parseInt(quantity) <= stockLevel
-
-    const isProductASet = product?.type.set
-
-    const renderActionButtons = () => {
-        const buttons = []
-        const buttonText = {
-            update: intl.formatMessage({
-                defaultMessage: 'Update',
-                id: 'product_view.button.update'
-            }),
-            addToCart: intl.formatMessage({
-                defaultMessage: 'Add to Cart',
-                id: 'product_view.button.add_to_cart'
-            }),
-            addSetToCart: intl.formatMessage({
-                defaultMessage: 'Add Set to Cart',
-                id: 'product_view.button.add_set_to_cart'
-            }),
-            addToWishlist: intl.formatMessage({
-                defaultMessage: 'Add to Wishlist',
-                id: 'product_view.button.add_to_wishlist'
-            }),
-            addSetToWishlist: intl.formatMessage({
-                defaultMessage: 'Add Set to Wishlist',
-                id: 'product_view.button.add_set_to_wishlist'
-            })
-        }
+const ProductView = forwardRef(
+    (
+        {
+            product,
+            category,
+            showFullLink = false,
+            imageSize = 'md',
+            isWishlistLoading = false,
+            addToCart,
+            updateCart,
+            addToWishlist,
+            updateWishlist,
+            isProductLoading,
+            isProductPartOfSet = false,
+            onVariantSelected = () => {},
+            validateAttributeSelection = (variant, quantity, stockLevel) =>
+                !isProductLoading && variant?.orderable && quantity > 0 && quantity <= stockLevel
```

Comment:
IMO `validateAttributeSelection` is a good name when used in the context of product sets (parent), because we're concerned with just the variant/attribute selection there.

But in this case, we're also checking the quantity and stock level too. So it's no longer just about the variant/attribute selection. 

So what would be a better name? 
- `validateOrderability`: I kinda like the original name `canOrder`
- `validateAddToCart`
- what else?

Reply:
The reason why I stepped away from `canOrder` was because it doesn't sound like a function, more so a boolean attribute that you set. I had to change it to be a function because we are using the productView in the pdp for the set and we need to customize the validation. So now we are stuff with coming up with a hard to name name. 

I don't want to spend that much time on naming conversations.. so you can pick the name. What do you want?

Reply:
In order to keep naming conversations to a minimum I've changed the function name to `validateOrderability`